extensions-contrib/opencensus-extensions/src/main/java/org/apache/druid/data/input/opencensus/protobuf/OpenCensusProtobufInputFormat.java

```diff
@@ -59,6 +69,35 @@ public boolean isSplittable()
   @Override
   public InputEntityReader createReader(InputRowSchema inputRowSchema, InputEntity source, File temporaryDirectory)
   {
+    try {
+      KafkaRecordEntity kafkaInputEntity = (KafkaRecordEntity) source;
+      Header versionHeader = kafkaInputEntity.getRecord().headers().lastHeader(VERSION_HEADER_KEY);
+      int version = 0;
+      if (versionHeader != null) {
+        version = ByteBuffer.wrap(versionHeader.value()).order(ByteOrder.LITTLE_ENDIAN).getInt();
+      }
+
+      if (version == 1) {
+        return new OpenTelemetryMetricsProtobufReader(
+          inputRowSchema.getDimensionsSpec(),
+          (ByteEntity) source,
+          metricDimension,
+          DEFAULT_VALUE_DIMENSION,
+          metricLabelPrefix,
+          resourceLabelPrefix
+        );
+      }
+    }
+    catch (ClassCastException e) {
```

Comment:
catching exceptions is expensive, instead we should do an `instanceof` check, which is much less onerous

Reply:
good point! updated as suggested